indico/modules/events/registration/badges.py

```diff
@@ -87,7 +87,12 @@ def _draw_badge(self, canvas, registration, template, tpl_data, pos_x, pos_y):
 
         placeholders = get_placeholders('designer-fields')
 
-        for item in tpl_data.items:
+        # Print images first
+        images = [key for key in placeholders if hasattr(placeholders[key], 'is_image')]
+        items = []
+        map(lambda x: items.insert(0, x) if x['type'] in images else items.append(x), tpl_data.items)
```

Comment:
Please don't use `map()` as a replacement for a for loop where you don't care about return values. This is not how `map()` is intended to be used and will cease to work in Python 3 where it returns an interator. What about something like this instead?

```python
image_placeholders = {name for name, placeholder in placeholders.viewitems() if placeholder.is_image}
image_items = [x for x in items if x['type'] in image_placeholders]
text_items = [x for x in items if x['type'] not in image_placeholders]

for item in chain(image_items, text_items):
    ...
```

Reply:
Actually, even easier:

```python
image_placeholders = {name for name, placeholder in placeholders.viewitems() if placeholder.is_image}
items = tpl_data.items.sort(key=lambda item: item['type'] not in image_placeholders)
for item in items:
    ...
```

This will sort anything that is in `image_placeholders` first.

Reply:
I really like this last version!
I didn't like the way I did it but I was not able to find a nicer way... this definitely nailed it :)

Reply:
Just one thing... the `for` should be using` tpl_data.items` :
`for item in tpl_data.items:`
because `items` is `None`

Reply:
I can even change this:
`items = tpl_data.items.sort(key=lambda item: item['type'] not in image_placeholders)`
to this:
`tpl_data.items.sort(key=lambda item: item['type'] not in image_placeholders)`
What do you think?

Reply:
sorry, i meant this of course. not sure why i went for the inplace sort (which is wrong here since you don't want to modify the original list!):

```python
items = sorted(tpl_data.items, key=lambda item: item['type'] not in image_placeholders)
```